src/Core/Settings.h

```diff
@@ -83,6 +83,7 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(UInt64, distributed_connections_pool_size, 1024, "Maximum number of connections with one remote server in the pool.", 0) \
     M(UInt64, connections_with_failover_max_tries, DBMS_CONNECTION_POOL_WITH_FAILOVER_DEFAULT_MAX_TRIES, "The maximum number of attempts to connect to replicas.", 0) \
     M(UInt64, s3_min_upload_part_size, 16*1024*1024, "The minimum size of part to upload during multipart upload to S3.", 0) \
+    M(UInt64, s3_max_upload_part_size, 5ull*1024*1024*1024, "The maximum size of part to upload during multipart upload to S3.", 0) \
```

Comment:
I'd better create a separate pr for this.

Reply:
> Parameters s3_truncate_on_insert, s3_create_new_file_on_insert, s3_check_objects_after_upload

Need not only mention in docs, but also explanation about what they do.
They alters behavior of ClickHouse during writing data to s3 and you also need to have different permissions in s3 for them. 
IE it can be confusing for people, why they need to have "delete" permission during write.